server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -159,4 +172,50 @@ private ImmutableList<Account> searchAccounts(
                                 searchParameters.nameQuery().get().toLowerCase(Locale.ROOT)))))
         .collect(ImmutableList.toImmutableList());
   }
+
+  /**
+   * This function updates the Applicant's date of birth by calling the UpdateApplicant() on the
+   * User Repository.
+   *
+   * @param trustedIntermediaryGroup - the TIGroup who manages the account whose Dob needs to be
+   *     updated.
+   * @param accountId - the account Id of the applicant whose Dob should be updated
+   * @param form - this contains the dob field which would be parsed into local date and updated for
+   *     the applicant
+   * @return form - the form object is always returned. If the form contains error, the controller
+   *     will handle the flash messages If the account is not found for the given AccountId, a
+   *     runtime exception is raised.
+   */
+  public Form<UpdateApplicantDobForm> updateApplicantDateOfBirth(
+      TrustedIntermediaryGroup trustedIntermediaryGroup,
+      Long accountId,
+      Form<UpdateApplicantDobForm> form) {
+
+    form = validateDateOfBirthForUpdateDob(form);
+
+    if (form.hasErrors()) {
+      return form;
+    }
+    Optional<Account> optionalAccount =
+        trustedIntermediaryGroup.getManagedAccounts().stream()
+            .filter(account -> account.id.equals(accountId))
+            .findAny();
+
+    if (optionalAccount.isEmpty() || optionalAccount.get().newestApplicant().isEmpty()) {
+      throw new RuntimeException(new ApplicantNotFoundException(accountId));
```

Comment:
Why a runtime exception? Can this throw ApplicantNotFoundException directly?

When that happens, the method will need to be declared to throw ApplicantNotFoundException. Then the top-level method in the controller `updateDateOfBirth` will also need to also declare that ApplicantNotFoundException is thrown.

Reply:
Changed it back to ApplicantNotFoundException.